packages/mux-player-react/src/lazy.tsx

```diff
@@ -1,117 +1,71 @@
-import React, { useEffect, useState } from 'react';
-import type { DetailedHTMLProps, HTMLAttributes } from 'react';
+import React, { useRef } from 'react';
 
 import ConditionalSuspense from './ConditionalSuspense';
 import useIsBrowser from './useIsBrowser';
 import useIsIntersecting from './useIsIntersecting';
 
 import type { MuxPlayerProps, MuxPlayerRefAttributes } from './index';
 
-interface MuxPlayerElement extends DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement> {
-  nohotkeys?: boolean | undefined;
-}
-
-declare global {
-  // eslint-disable-next-line @typescript-eslint/no-namespace
-  namespace JSX {
-    interface IntrinsicElements {
-      'mux-player': MuxPlayerElement;
-    }
-  }
-}
-
 const MuxPlayerIndex = React.lazy(() => import('./index'));
 
-interface FallbackProps extends MuxPlayerProps {
-  onIntersection?: () => void;
+interface MuxPlayerLazyProps extends MuxPlayerProps {
+  loading?: 'page' | 'viewport';
 }
-const Fallback = (props: FallbackProps) => {
-  const { style, className, onIntersection, placeholder } = props;
+const MuxPlayer = React.forwardRef<MuxPlayerRefAttributes, MuxPlayerLazyProps>((props, ref) => {
+  const { loading = 'viewport', ...playerProps } = props;
 
-  const intersectionRef = React.useRef<HTMLElement>(null);
+  // We load mux player once two conditions are met:
+  // 1. We're in a browser (react.lazy doesn't work on the server in react 17)
+  const isBrowser = useIsBrowser();
+  // 2. The player has entered the viewport, according to the fallback (if enabled).
+  const intersectionRef = useRef<HTMLDivElement>(null);
   const isIntersecting = useIsIntersecting(intersectionRef);
 
-  useEffect(() => {
-    if (isIntersecting && onIntersection) {
-      onIntersection();
-    }
-  }, [isIntersecting, onIntersection]);
-
   return (
-    /* 
-    Why do we have a mux-player element before the mux-player bundle is even loaded?
-    Before the bundle is loaded, this mux-player element just acts like a div.
-    However, by calling this placeholder "mux-player",
-    it now gets the same CSS applied to it that the eventual "real" mux-player element will. 
-    */
     <>
-      <mux-player
-        ref={intersectionRef}
-        data-mux-player-react-lazy-placeholder
-        placeholder={placeholder}
-        style={style}
-        className={className || ''}
-        // since there's a possibility that the bundle loads before Suspense clears this placeholder,
-        // we need to make sure that the placeholder isn't interactive and its player chrome in doesn't get rendered
-        nohotkeys
-        aria-hidden
-        tabIndex={-1}
-      >
-        <div data-mux-player-react-lazy-placeholder-overlay />
-      </mux-player>
+      <div data-mux-player-react-lazy ref={intersectionRef}>
+        <ConditionalSuspense
+          condition={isBrowser && (isIntersecting || loading === 'page')}
```

Comment:
suggestion(non-blocking): related to the above suggestion, we should change this `||` order if we can avoid computing `isIntersecting` for `loading === 'page'`